client/odie/send-message-input/index.tsx

```diff
@@ -31,12 +35,51 @@ export const OdieSendMessageButton = ( {
 	const dispatch = useDispatch();
 	const translate = useTranslate();
 
+	const { siteId, siteDomain, currentRoute, currentPlan } = useSelector( ( state ) => {
+		const siteId = getSelectedSiteId( state ) as number;
+		const siteDomain = getSiteDomain( state, siteId ) as string;
+		const currentRoute = getCurrentRoute( state );
+		const currentPlan = getCurrentPlan( state, siteId );
+
+		return {
+			siteId,
+			siteDomain,
+			currentRoute,
+			currentPlan,
+		};
+	} );
+
 	useEffect( () => {
 		if ( initialUserMessage && ! chat.chat_id ) {
 			setMessageString( initialUserMessage );
 		}
 	}, [ initialUserMessage, chat.chat_id ] );
 
+	/**
+	 * Replaces route param values with placeholders, and returns the modified route path and route params.
+	 * @param  {string}  route The route path
+	 * @returns {string, Object} Object containing the modified route path and route params
+	 */
+	const replaceRouteParamsWithPlaceholders = ( route: string ) => {
+		const routeParams = {} as Record< string, string >;
+
+		route = route.replace( siteDomain, ':site' );
+		routeParams.site = siteDomain;
+
+		return { route, routeParams };
+	};
+
+	const buildContext = () => {
+		const { route, routeParams } = replaceRouteParamsWithPlaceholders( currentRoute );
+
+		return {
+			blog_id: siteId,
+			route: route,
+			route_params: routeParams,
+			site_plan_id: currentPlan?.id ?? null,
```

Comment:
Let's remove `site_plan_id`. I'll just add it on the backend based on the `blog_id`.